src/Core/src/Platform/Android/ImageViewExtensions.cs

```diff
@@ -0,0 +1,105 @@
+﻿#nullable enable
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Android.Graphics.Drawables;
+using Android.Widget;
+
+namespace Microsoft.Maui
+{
+	public static class ImageViewExtensions
+	{
+		public static void Clear(this ImageView imageView)
+		{
+			// stop the animation
+			if (imageView.Drawable is IAnimatable animatable)
+				animatable.Stop();
+
+			// clear the view and release any bitmaps
+			imageView.SetImageResource(global::Android.Resource.Color.Transparent);
+		}
+
+		public static void UpdateAspect(this ImageView imageView, IImage image)
+		{
+			imageView.SetScaleType(image.Aspect.ToScaleType());
+		}
+
+		public static void UpdateIsAnimationPlaying(this ImageView imageView, IImageSourcePart image)
+		{
+			if (imageView.Drawable is IAnimatable animatable)
+			{
+				if (image.IsAnimationPlaying)
+				{
+					if (!animatable.IsRunning)
+						animatable.Start();
+				}
+				else
+				{
+					if (animatable.IsRunning)
+						animatable.Stop();
+				}
+			}
+		}
+
+		public static async Task<IImageSourceServiceResult<Drawable>?> UpdateSourceAsync(this ImageView imageView, IImageSourcePart image, IImageSourceServiceProvider services, CancellationToken cancellationToken = default)
+		{
+			imageView.Clear();
+
+			image.UpdateIsLoading(false);
+
+			var context = imageView.Context;
+			if (context == null)
+				return null;
+
+			var imageSource = image.Source;
+			if (imageSource == null)
+				return null;
+
+			var events = image as IImageSourcePartEvents;
+
+			events?.LoadingStarted();
+			image.UpdateIsLoading(true);
+
+			try
+			{
+				var service = services.GetRequiredImageSourceService(imageSource);
+
+				var result = await service.GetDrawableAsync(imageSource, context, cancellationToken);
+				var drawable = result?.Value;
+
+				var applied = !cancellationToken.IsCancellationRequested && imageView.IsAlive() && imageSource == image.Source;
+
+				// only set the image if we are still on the same one
+				if (applied)
+				{
+					imageView.SetImageDrawable(drawable);
```

Comment:
I realize the cancellationtokens do things but I feel like still having some checks here to validate the imageview hasn't been disposed and has a handle would be useful

Reply:
I did, in the live above:

```csharp
var applied = !cancellationToken.IsCancellationRequested && imageView.IsAlive() && imageSource == image.Source;
```